src/riak_ensemble_peer.erl

```diff
@@ -1802,6 +1844,18 @@ setup({init, Args}, State0=#state{id=Id, ensemble=Ensemble, ets=ETS, mod=Mod}) -
     probe(init, State3).
 
 -spec handle_event(_, atom(), state()) -> {next_state, atom(), state()}.
+handle_event({watch_leader_status, Pid}, StateName, State) when node(Pid) =/= node() ->
+    lager:warning("Remote pid ~p not allowed to watch_leader_status on ensemble peer ~p",
+                  [Pid, State#state.id]),
+    {next_state, StateName, State};
+handle_event({watch_leader_status, Pid}, StateName, State = #state{watchers = Watchers}) ->
+    _ = notify_leader_status(Pid, StateName, State),
+    %% Might as well take this opportunity to prune any dead pids that are in the list
+    NewWatcherList = [P || P <- [Pid | Watchers], is_process_alive(P)],
```

Comment:
Yeah, so at the time that I implemented this, it turned out to be a lot easier to do it this way, because the existing code does some weird stuff with monitors where DOWN signals get passed through to the ensemble backend module. We could still use monitors, but it would require saving some extra stuff in the state so that we can distinguish DOWN signals from watchers vs DOWN signals intended for the backend.

Since doing that would require even more code, and have the additional overhead of creating extra monitors, I opted to give this approach a try. I agree it's not terribly idiomatic though, so I could probably be persuaded pretty easily to rewrite this to use monitors. Let me know what you think.